guillotina/fields/annotation.py

```diff
@@ -277,25 +277,33 @@ def get_annotation_name(self, bucket_id: str) -> str:
             # get annotation for this key again as it might be the new annotation
             annotation = await self.get_annotation(context, key)
 
-        if key in annotation["keys"]:
+        insert_idx = bisect.bisect_left(annotation["keys"], key)
+        if len(annotation["keys"]) > insert_idx and annotation["keys"][insert_idx] == key:
             # change existing value
-            idx = annotation["keys"].index(key)
-            annotation["values"][idx] = value
+            annotation["values"][insert_idx] = value
         else:
-            insert_idx = bisect.bisect_left(annotation["keys"], key)
             annotation["keys"].insert(insert_idx, key)
             annotation["values"].insert(insert_idx, value)
             _, bucket = self._find_bucket(key)
             bucket["len"] = len(annotation["keys"])
 
         annotation.register()
 
+    def __index(self, annotation, key):
+        idx = bisect.bisect_left(annotation["keys"], key)
+        try:
+            if annotation["keys"][idx] == key:
+                return idx
+        except IndexError:
+            pass
+        raise ValueError(key)
```

Comment:
Shouldn't we provide BBB ?

Reply:
What would be bbb be needed for?

Reply:
we are only changing lookups from [].index to bisect

Reply:
```
>>> annotations = ['myannotation1', 'myannotation5', 'myannotation3']
>>> bisect.bisect_left(annotations, 'myannotation3')
1
>>> annotations.index('myannotation3')
2
```

Reply:
![image](https://user-images.githubusercontent.com/775189/74336181-19213680-4d9e-11ea-86b9-313aa5ba2544.png)

Reply:
@bloodbare I don't understand. This is internal implementation for inserting/updating. No apis have changed.